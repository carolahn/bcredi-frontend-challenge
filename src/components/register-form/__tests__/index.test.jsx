import React from "react";
import renderer from "react-test-renderer";

import RegisterForm from "../index";

jest.mock("react-hook-form", () => ({
  useForm: () => ({
    register: jest.fn(),
    handleSubmit: jest.fn((func) => func),
    errors: {},
    setError: jest.fn(),
  }),
}));

describe("register-form render test", () => {
  it("renders correctly RegisterForm", () => {
    const tree = renderer
      .create(
        <RegisterForm
          values={{}}
          onSubmit={() => {}}
          formErrors={{
            register: () => {},
            handleSubmit: () => {},
            errors: {},
          }}
          handleOnChange={() => {}}
          applyMask={() => {}}
          togglePassword={() => {}}
          showPassword={() => {}}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
