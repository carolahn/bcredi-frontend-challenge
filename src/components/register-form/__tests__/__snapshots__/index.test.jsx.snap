// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`register-form render test renders correctly RegisterForm 1`] = `
<form
  className="sc-bdfBwQ gBrTql"
  name="register"
>
  <div
    className="register__input-wrapper"
  >
    <label
      className="register__label"
    >
      E-mail
    </label>
    <input
      className="sc-gsTCUz fKZdNo register__input"
      name="email"
      onChange={[Function]}
      placeholder="exemplo@email.com"
      style={
        Object {
          "border": "1px solid #e6e6e6",
          "color": "#333333",
        }
      }
      width="320px"
    />
  </div>
  <div
    className="register__container"
  >
    <div
      className="register__small-input-container"
    >
      <label
        className="register__label register__label--small"
      >
        CPF
      </label>
      <input
        className="sc-gsTCUz iyuuZk register__input--small"
        maxLength={18}
        name="cpf"
        onChange={[Function]}
        placeholder="000.000.000-00"
        style={
          Object {
            "border": "1px solid #e6e6e6",
            "color": "#333333",
          }
        }
        width="148px"
      />
    </div>
    <div
      className="register__small-input-container"
    >
      <label
        className="register__label register__label--small"
      >
        Data de nascimento
      </label>
      <input
        className="sc-gsTCUz iyuuZk register__input--small"
        maxLength={10}
        name="birth"
        onChange={[Function]}
        placeholder="DD/MM/AAAA"
        style={
          Object {
            "border": "1px solid #e6e6e6",
            "color": "#333333",
          }
        }
        width="148px"
      />
    </div>
  </div>
  <div
    className="register__input-wrapper"
  >
    <label
      className="register__label"
    >
      Senha
    </label>
    <div
      className="register__input--toggle-eye"
    >
      <input
        className="sc-gsTCUz fKZdNo register__input"
        name="password"
        onChange={[Function]}
        placeholder="Cadastre uma senha"
        style={
          Object {
            "border": "1px solid #e6e6e6",
            "color": "#333333",
          }
        }
        type="text"
        width="320px"
      />
      <i
        className="register__icon--eye"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-eye fa-w-18 "
          data-icon="eye"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M572.52 241.4C518.29 135.59 410.93 64 288 64S57.68 135.64 3.48 241.41a32.35 32.35 0 0 0 0 29.19C57.71 376.41 165.07 448 288 448s230.32-71.64 284.52-177.41a32.35 32.35 0 0 0 0-29.19zM288 400a144 144 0 1 1 144-144 143.93 143.93 0 0 1-144 144zm0-240a95.31 95.31 0 0 0-25.31 3.79 47.85 47.85 0 0 1-66.9 66.9A95.78 95.78 0 1 0 288 160z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </i>
    </div>
  </div>
  <div
    className="sc-hKgILt eoiDfF"
    style={
      Object {
        "border": "1px solid #e6e6e6",
      }
    }
  >
    <input
      className="register__input--checkbox"
      name="agree"
      onChange={[Function]}
      style={
        Object {
          "border": "1px solid #e6e6e6",
          "color": "#333333",
        }
      }
      type="checkbox"
      width="320px"
    />
    <span
      className="register__checkbox-label"
    >
      Li e estou de acordo com a
       
      <a
        href="#politica-de-privacidade"
        title="Política de privacidade da Bcredi"
      >
        Política de Privacidade
      </a>
       
      e a
       
      <a
        href="#politica-de-uso"
        title="Política de uso da Bcredi"
      >
        Política de Uso de Informações
      </a>
      .
    </span>
  </div>
  <button
    className="sc-eCssSg gYctEH"
  >
    <i
      className="register__icon--lock"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-lock fa-w-14 "
        data-icon="lock"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M400 224h-24v-72C376 68.2 307.8 0 224 0S72 68.2 72 152v72H48c-26.5 0-48 21.5-48 48v192c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V272c0-26.5-21.5-48-48-48zm-104 0H152v-72c0-39.7 32.3-72 72-72s72 32.3 72 72v72z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </i>
    Cadastrar
  </button>
</form>
`;
